track.py

```diff
@@ -61,12 +62,38 @@ class Track:
     @staticmethod
     def from_xml_file(filename):
         xmldoc = parse(filename)
+        width = get_track_root_info(xmldoc, 'width')
+        profil_steps_length = get_track_root_info(xmldoc, 'profil steps length')
+
         segments = []
         for section in get_main_track(xmldoc).childNodes:
             if section.nodeType == Node.ELEMENT_NODE:
-                segments.append(Segment.from_xml(section))
+                s = Segment.from_xml(section)
+                if s.end_radius is None:
+                    segments.append(s)
+                else:
+                    length = abs((s.radius + s.end_radius)/2.0 * s.arc)
+                    num_steps = int(length/4.0) + 1
+                    if num_steps == 1:
+                        segments.append(s)
+                    else:
+                        dlen = length/float(num_steps - 1)
+                        dradius = (s.end_radius - s.radius)/float(num_steps - 1)
+
+                        tmp_angle = 0.0
+                        for i in xrange(num_steps):
+                            tmp_angle += dlen/(s.radius + i*dradius)
+                        dlen *= s.arc / tmp_angle
```

Comment:
This is "imitation" (~ copy & paste) of original Speed Dreams code [track4.cpp#L1445-L1454](https://github.com/robotika/Roborace-Dreams/blob/56a8771d1ab385c4c81677ee66a46686749b3135/src/modules/track/trackv1/track4.cpp#L1445-L1454)